builtins.go

```diff
@@ -248,8 +249,79 @@ func builtinType(e *evaluator, xp potentialValue) (value, error) {
 	return makeValueString(x.typename()), nil
 }
 
+func makeDoubleCheck(e *evaluator, x float64) (value, error) {
+	if math.IsNaN(x) {
+		return nil, e.Error("Not a number")
+	}
+	if math.IsInf(x, 0) {
+		return nil, e.Error("Overflow")
+	}
+	return makeValueNumber(x), nil
+}
+
+// TODO(sbarzowski) perhaps it is too magical for Go style
```

Comment:
I like it

Reply:
these kinds of abstractions are basically mandatory given the error boilerplate